mass_mailing_sending_queue/models/mail_mass_mailing_sending.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from openerp import api, fields, models, tools
+
+_logger = logging.getLogger(__name__)
+
+
+class MailMassMailingSending(models.Model):
+    _name = 'mail.mass_mailing.sending'
+
+    state = fields.Selection([
+        ('draft', "Draft"),
+        ('enqueued', "Enqueued"),
+        ('sending', "Sending"),
+        ('sent', "Sent"),
+        ('error', "Error"),
+    ], string="State", required=True, copy=False, default='enqueued')
+    mass_mailing_id = fields.Many2one(
+        string="Mass mailing", comodel_name='mail.mass_mailing', readonly=True)
+    pending = fields.Integer(
+        string="Recipients pending", compute='_compute_pending')
+    sending = fields.Integer(
+        string="Emails sending", compute='_compute_sending')
+    sent = fields.Integer(
+        string="Emails sent", compute='_compute_sent')
+    failed = fields.Integer(
+        string="Emails failed", compute='_compute_failed')
+    error = fields.Char(string="Error message")
+    date_start = fields.Datetime(
+        string="Date start", default=fields.Datetime.now())
+    date_end = fields.Datetime(string="Date end")
+
+    def _batch_size_default(self):
+        return 500
```

Comment:
Why this method? I prefer to hardcode it in the other one. Inheratibility is not a problem here, because you can easily overwrite XML if you want another value in your child module.